geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java

```diff
@@ -877,14 +879,32 @@ public void init(Services<ID> s) throws MembershipConfigurationException {
   @Override
   public void started() throws MemberStartupException {
     setLocalAddress(services.getMessenger().getMemberID());
+    createServerSocket();
+    startTcpServer(serverSocket);
+    startHeartbeatThread();
+  }
+
+  private void createServerSocket() throws MemberStartupException {
     try {
-      serverSocket = createServerSocket(localAddress.getInetAddress(),
-          services.getConfig().getMembershipPortRange());
+      InetAddress address = localAddress.getInetAddress();
+      String bindAddrStr = services.getConfig().getBindAddress();
+      if (bindAddrStr != null && !bindAddrStr.isEmpty()) {
+        try {
+          if (LocalHostUtil.isWildcardAddress(bindAddrStr)) {
+            address = LocalHostUtil.getAnyLocalAddress();
+          } else {
+            address = InetAddress.getByName(bindAddrStr);
+          }
+        } catch (UnknownHostException e) {
+          logger.error(
+              "Error when configuring {} as bind address in membership, default address will be used. Exception: {}",
+              services.getConfig().getBindAddress(), e.getMessage());
```

Comment:
I was going to recommend logging the full exception but I feel like `e.getMessage()` is sufficient and reduces log clutter ✓